stl/inc/algorithm

```diff
@@ -10102,9 +10102,14 @@ namespace ranges {
             if constexpr (forward_range<_Rng> && _Prefer_iterator_copies<iterator_t<_Rng>>) {
                 const auto _Found = _RANGES _Minmax_element_unchecked(
                     _STD move(_UFirst), _STD move(_ULast), _Pass_fn(_Pred), _Pass_fn(_Proj));
+                if (_Found.min == _Found.max) {
+                    _Vty _Temp{*_Found.min};
+                    return {_Temp, _Temp};

```

Comment:
This does two copies plus a move (or three copies in the `move_iterator` case). Given that we can't be clever and avoid a move when `*_Found.min` is not a reference (i.e., `static_cast<_Vty>(static_cast<const _Vty&>(*_Found.min))`), I believe we should instead make sure to `move(_Temp)` for the second element. Since it's in `{}`, the ordering will work out correctly and initialize with the copy of `_Temp` for `min`, followed by a move of `_Temp` for `max`.

Reply:
Does `{}` guarantee ordering unlike `()`?

Reply:
Braces do guarantee ordering; MSVC emits a warning when compiler bugs prevent it from following the Standard (last I checked). I worry somewhat that this is overly clever.

Reply:
After reading the code, I think that this is fine - the named variable makes it sufficiently obvious what is happening, and the line isn't too complicated.